### PR TITLE
Stop Generating MessageId

### PR DESCRIPTION
**Pull request by jlee9595**
Accompanies https://github.com/segmentio/analytics.js-core/pull/106

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -12,7 +12,6 @@ var integration = require('@segment/analytics.js-integration');
 var json = require('json3');
 var keys = require('@ndhoule/keys');
 var localstorage = require('yields-store');
-var md5 = require('spark-md5').hash;
 var protocol = require('@segment/protocol');
```
Comment (f2prateek) [inline lib/index.js:15]: do we still need the uuid dependency?

Comment (jlee9595) [reply lib/index.js:15]: yep see https://github.com/segment-integrations/analytics.js-integration-segmentio/pull/55/files#diff-6d186b954a58d5bb740f73d84fe39073R477

Comment (f2prateek) [reply lib/index.js:15]: ah ok 👍

**Comment by f2prateek** [comment]
Code LGTM!

I think  we should update the version of a.js core as a dev here and add a test to verify it picks the messageId from whatever a.js provides.

One more thing we can consider to reduce risk is to leave the generation logic as is, but make it check the root message first.

`msg.messageId = msg.messageId || 'ajs-' + md5(json.stringify(msg) + uuid());`

File: test/index.test.js [hunk 2]
```diff
@@ -425,6 +410,13 @@ describe('Segment.io', function() {
           assert(!object._metadata);
         });
       });
+
+      it('should pick up messageId from AJS', function() {
```
Comment (jlee9595) [inline test/index.test.js:414]: @f2prateek Is this along the lines of what you were looking for?

Comment (f2prateek) [reply test/index.test.js:414]: Yep!

**Review by f2prateek — APPROVED** [review]

**Comment by f2prateek** [comment]
(looks like some lint errors were flagged in CI, let's get those fixed before merging too)

**Pull request merged**